### ISSUE TITLE
API: Force Sync not working

### ISSUE THREAD

**Issue by rhenretta** [Shippable/support#3747]
I’m writing an application that creates a repo in github, then adds boilerplate CI/CD through shippable.  Since the repo was literally just created, I force a refresh on shippable.

 

First I get all accounts:

```
Making GET request to /accounts? with data: {}
[ { id: [snip],
lastSyncStartDate: '2017-08-01T03:06:17.260Z',
lastSyncEndDate: '2017-08-01T03:06:26.885Z',
defaultEmail: [snip],
hubspotId: [snip],
clearbitProfileId: [snip],
createdAt: '2017-02-24T15:46:43.393Z',
isSyncing: false,
lastLoginAt: '2017-07-18T20:37:08.500Z',
defaultViewId: null } ]
```

Then I force a refresh:
```
Making GET request to /accounts/[snip]?forceSync=true with data: {"forceSync":true}
{ id: [snip],
accountId: [snip],
braintreeCustomerId: null,
lastSyncEndDate: '2017-08-01T03:06:26.885Z',
lastSyncStartDate: '2017-08-01T03:06:17.260Z',
defaultEmail: [snip],
hubspotId: [snip],
clearbitProfileId: [snip],
firstLoginAt: '2017-02-24T15:46:43.392Z',
lastLoginAt: '2017-07-18T20:37:08.500Z',
lastHubspotSyncAt: '2017-08-01T03:06:27.216Z',
isSyncing: false,
defaultViewId: null,
npsLatest: null,
npsCount: null,
npsAverage: null,
npsUpdatedAt: null,
createdAt: '2017-02-24T15:46:43.393Z',
updatedAt: '2017-08-01T03:06:27.317Z',
isSyncPending: true }
```

At this point, I refresh the account every 5 seconds until I confirm the refresh is complete.  I confirm by checking IsSyncPending, isSyncing, and lastSyncEndDate:
```
Making GET request to /accounts/[snip]? with data: {}
{ id: [snip],
accountId: [snip],
braintreeCustomerId: null,
lastSyncEndDate: '2017-08-01T03:06:51.597Z',
lastSyncStartDate: '2017-08-01T03:06:45.101Z',
defaultEmail: [snip],
hubspotId: [snip],
clearbitProfileId: [snip],
firstLoginAt: '2017-02-24T15:46:43.392Z',
lastLoginAt: '2017-07-18T20:37:08.500Z',
lastHubspotSyncAt: '2017-08-01T03:06:51.729Z',
isSyncing: false,
defaultViewId: null,
npsLatest: null,
npsCount: null,
npsAverage: null,
npsUpdatedAt: null,
createdAt: '2017-02-24T15:46:43.393Z',
updatedAt: '2017-08-01T03:06:51.771Z',
isSyncPending: false }
Sync complete
```
 
The next call is to find the new project, but it returns empty:
```
Making GET request to /projects?name=hello-shippable with data: {"name":"hello-shippable"}
[]
```
 
Even several minutes later, it still isn’t showing up:
```
$ date
Mon Jul 31 22:15:46 CDT 2017
$ curl -X GET --header 'Accept: application/json' --header 'Authorization: apiToken [snip]’ 'https://api.shippable.com/projects?name=hello-shippable'
[]
```

However, when I do a refresh on your site the project is immediately available:
```
$ date
Mon Jul 31 22:17:18 CDT 2017
$ curl -X GET --header 'Accept: application/json' --header 'Authorization: apiToken [snip]’ 'https://api.shippable.com/projects?name=hello-shippable'

[{"branches":null,"sourceRepoOwner":{"login":[snip],"id":[snip],"avatar_url":"https://avatars1.githubusercontent.com/u/[snip]?v=4","gravatar_id":"","url":"https://api.github.com/users/[snip]","html_url":"https://github.com/[snip]","followers_url":"https://api.github.com/users/[snip]/followers","following_url":"https://api.github.com/users/[snip]/following{/other_user}","gists_url":"https://api.github.com/users/[snip]/gists{/gist_id}","starred_url":"https://api.github.com/users/[snip]/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/[snip]/subscriptions","organizations_url":"https://api.github.com/users/[snip]/orgs","repos_url":"https://api.github.com/users/[snip]/repos","events_url":"https://api.github.com/users/[snip]/events{/privacy}","type":"Organization","site_admin":false},"propertyBag":{"repositoryHttpsUrl":"https://github.com/[snip]/hello-shippable.git"},"id":"[snip]","projectId":[snip],"name":"hello-shippable","fullName":"[snip]/hello-shippable","subscriptionId":"[snip]","ownerAccountId":null,"builderAccountId":null,"repositoryUrl":"https://api.github.com/repos/[snip]/hello-shippable","repositorySshUrl":"git@github.[snip]/hello-shippable.git","repositoryHtmlUrl":"https://github.com/[snip]/hello-shippable","customCloneUrl":null,"sourceDefaultBranch":null,"isPrivateRepository":true,"isOrg":true,"isFork":false,"sourceId":"[snip]","sourcePushedAt":"2017-08-01T03:06:41.000Z","sourceCreatedAt":"2017-08-01T03:06:35.000Z","sourceUpdatedAt":"2017-08-01T03:06:35.000Z","lastBuildGroupNumber":null,"autoBuild":null,"deployPublicKey":null,"language":null,"providerId":"[snip]","providerLastSyncEndDate":null,"providerLastSyncStartDate":null,"enabledDate":null,"enabledBy":null,"enabledByUserName":null,"type":null,"consolidateReports":false,"isOrphaned":false,"createdBy":"[snip]","updatedBy":"[snip]","createdAt":"2017-08-01T03:17:10.740Z","updatedAt":"2017-08-01T03:17:10.748Z"}]

**Comment by rageshkrishna** [comment]
Hi @rhenretta. This seems very strange. Our UI doesn't do anything more than the same API call to sync your account, and I've been unable to reproduce the problem at my end. Here's what I've tried:

1. Login to Shippable
1. Call GET /projects?name=test-update. Verified that I get an empty array
1. Go to GitHub and create a brand new repo
1. Call GET /accounts/{id}?forceSync=true
    Note that no request body with `{ "forceSync": true }` is necessary for this, but including it should make no difference at all
1. Call GET /projects?name=test-update. Verified that I got a project.

Based on your description, this is exactly what you've done so I'm not really sure why it didn't work. Does this happen to you consistently?
